code/game/mecha/equipment/tools/work_tools.dm

```diff
@@ -30,28 +30,38 @@
 		return
 	if(!cargo_holder)
 		return
-	if(istype(target,/obj))
+	if(isobj(target))
 		var/obj/O = target
-		if(!O.anchored)
-			if(cargo_holder.cargo.len < cargo_holder.cargo_capacity)
-				chassis.visible_message("[chassis] lifts [target] and starts to load it into cargo compartment.")
-				O.anchored = 1
-				if(do_after_cooldown(target))
-					cargo_holder.cargo += O
-					O.loc = chassis
-					O.anchored = 0
-					occupant_message("<span class='notice'>[target] successfully loaded.</span>")
-					log_message("Loaded [O]. Cargo compartment capacity: [cargo_holder.cargo_capacity - cargo_holder.cargo.len]")
-				else
-					O.anchored = initial(O.anchored)
-			else
-				occupant_message("<span class='warning'>Not enough room in cargo compartment!</span>")
-		else
+		if(istype(target, /obj/machinery/power/supermatter_crystal)) //No, you can't pick up the SM with this you moron, did you think you were clever?
+			var/obj/mecha/working/ripley/R = chassis
+			QDEL_LIST(R.cargo) //We don't want to drop cargo that just spam hits the SM, let's delete it
+			occupant_message("<span class='userdanger'>You realise in horror what you have done as [chassis] starts warping around you!</span>")
+			chassis.occupant.dust()
```

Comment:
So.. this would most likely dust an AI remotely controlling the Ripley, would it not?

Reply:
AIs that enter mechs cannot leave. There is no "remote control."
To recover an AI from a destroyed mech, you card it, and the mech will ALWAYS dust during this interaction. So dusting the AI is fine.